deb/openmediavault/usr/share/php/openmediavault/functions.inc

```diff
@@ -234,11 +234,12 @@ function array_rand_value(array $array, $num = 1) {
  * @ingroup api
  * @param array $array An array with keys.
  * @param string $key The key of the element.
+ * @param string $default The default value returned if not found.
  * @return Returns the boolean value of the given key.
  */
-function array_boolval(array $array, $key) {
+function array_boolval($array, $key, $default = false) {
```

Comment:
Any reason why you removed the type hint?

Reply:
try that command

`omv-rpc 'FileSystemMgmt' 'enumerateMountedFilesystems'`

you will get that error

```
May 31 16:39:30 proliant omv-engined[31288]: PHP Fatal error: Uncaught TypeError: Argument 1 passed to array_boolval() must be of the type array, null given, called in /usr/share/openmediavault/engined/rpc/filesystemmgmt.inc on line 213 and defined in /usr/share/php/openmediavault/functions.inc:239
May 31 16:39:30 proliant omv-engined[31288]: Stack trace:
May 31 16:39:30 proliant omv-engined[31288]: #0 /usr/share/openmediavault/engined/rpc/filesystemmgmt.inc(213): array_boolval(NULL, 'includeroot', false)
May 31 16:39:30 proliant omv-engined[31288]: #1 [internal function]: Engined\Rpc\OMVRpcServiceFileSystemMgmt->enumerateMountedFilesystems(NULL, Array)
May 31 16:39:30 proliant omv-engined[31288]: #2 /usr/share/php/openmediavault/rpc/serviceabstract.inc(123): call_user_func_array(Array, Array)
May 31 16:39:30 proliant omv-engined[31288]: #3 /usr/share/php/openmediavault/rpc/rpc.inc(86): OMV\Rpc\ServiceAbstract->callMethod('enumerateMounte...', NULL, Array)
May 31 16:39:30 proliant omv-engined[31288]: #4 /usr/sbin/omv-engined(537): OMV\Rpc\Rpc::call('FileSystemMgmt', 'enumerateMounte...', NULL, Array, 1)
May 31 16:39:30 proliant omv-engined[31288]: #5 {main}
```

in fact if array_boolval is called without params, $params get the NULL value and as the type expected is array in the function, we have the exception raised above.

The easiest way to remove the bug is to remove type in the expected parameters of that function.
And you did that for the $key, so I did it for $params. :)